src/main/java/edu/cornell/kfs/module/ar/service/impl/CuContractsGrantsBillingUtilityServiceImpl.java

```diff
@@ -0,0 +1,87 @@
+package edu.cornell.kfs.module.ar.service.impl;
+
+import java.sql.Date;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.kuali.kfs.coa.businessobject.Account;
+import org.kuali.kfs.coa.service.AccountService;
+import org.kuali.kfs.coreservice.framework.parameter.ParameterService;
+import org.kuali.kfs.integration.cg.ContractsAndGrantsBillingAwardAccount;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.module.ar.ArConstants;
+import org.kuali.kfs.module.ar.service.impl.ContractsGrantsBillingUtilityServiceImpl;
+import org.kuali.kfs.sys.KFSConstants;
+
+import edu.cornell.kfs.module.ar.CuArParameterKeyConstants;
+import edu.cornell.kfs.module.ar.service.CuContractsGrantsBillingUtilityService;
+import edu.cornell.kfs.sys.CUKFSParameterKeyConstants;
+
+public class CuContractsGrantsBillingUtilityServiceImpl extends ContractsGrantsBillingUtilityServiceImpl implements CuContractsGrantsBillingUtilityService {
+    
+    protected AccountService accountService;
+    protected ParameterService parameterService;
+    
+    @Override
+    public Date determineLastBilledDateByInvoicingOption(List<ContractsAndGrantsBillingAwardAccount> awardAccounts, String invoicingOptionCode, Date awardLastBilledDate) {
+        Date computedLastBilledDate = null;
+        
+        if (StringUtils.equalsIgnoreCase(ArConstants.INV_ACCOUNT, invoicingOptionCode)
+                || StringUtils.equalsIgnoreCase(ArConstants.INV_CONTRACT_CONTROL_ACCOUNT, invoicingOptionCode)) {
+            ContractsAndGrantsBillingAwardAccount accountToUse = null;
+            for (ContractsAndGrantsBillingAwardAccount awardAccount : awardAccounts) {
+                if (ObjectUtils.isNotNull(accountToUse)) {
+                    if (isNotExpenditureAccount(awardAccount) 
+                            && awardAccount.getCurrentLastBilledDate().after(accountToUse.getCurrentLastBilledDate())) {
+                        accountToUse = awardAccount;
+                    }
+                } else {
+                    accountToUse = awardAccount;
```

Comment:
Is it okay for the accountToUse to be an expenditure account if that's all there is in the list? Or is it critical that no expenditure accounts are used at all?